packages/restate-xstate/src/lib/actorObject.ts

```diff
@@ -171,15 +180,34 @@ export function actorObject<
           ctx.set("events", events);
         }
 
-        const root = (
-          await createActor<PreviousStateMachine | LatestStateMachine>(
-            ctx,
-            api,
-            systemName,
-            version,
-            logic,
-          )
-        ).start();
+        const root = await createActor<
+          PreviousStateMachine | LatestStateMachine
+        >(ctx, api, systemName, version, logic);
+
+        const subscriptions = (await ctx.get("subscriptions")) ?? {};
+
+        if (
+          request.subscribe &&
+          !evaluateCondition(ctx, root, request.subscribe.condition, [
+            request.subscribe.awakeableId,
+          ])
+        ) {
+          if (subscriptions[request.subscribe.condition]) {
```

Comment:
Here is probably best is to extract a variable, then no need for the `?` (which is confusing at first)